src/cli/circom.rs

```diff
@@ -0,0 +1,118 @@
+use std::{fs, path::Path, process::Command};
+
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+
+use ansi_term::Colour::{Green, Red};
+use anyhow::{bail, Result};
+use camino::Utf8PathBuf;
+
+use crate::cli::paths::{circom_binary_path, circom_dir};
+
+const CIRCOM_VERSION: &str = "2.1.6";
+
+#[cfg(target_arch = "wasm32")]
+fn download_circom_binary(_path: impl AsRef<Path>) -> Result<Command> {
+    bail!("wasm does not support downloading")
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn download_circom_binary(path: impl AsRef<Path>) -> Result<Command> {
+    use std::io::Write;
+
+    let url = match std::env::consts::OS {
+        "linux" => "https://github.com/iden3/circom/releases/download/v2.1.6/circom-linux-amd64",
+        "macos" => "https://github.com/iden3/circom/releases/download/v2.1.6/circom-macos-amd64",
+        "windows" => {
+            "https://github.com/iden3/circom/releases/download/v2.1.6/circom-windows-amd64.exe"
+        }
+        _ => {
+            eprintln!("Unsupported OS");
+            std::process::exit(1);
+        }
+    };
+
+    let response = reqwest::blocking::get(url)?.bytes()?;
+    let mut out = fs::File::create(path.as_ref())?;
+    out.write_all(&response)?;
+
+    #[cfg(unix)]
+    fs::set_permissions(path.as_ref(), fs::Permissions::from_mode(0o755))?;
+
+    Ok(Command::new(path.as_ref().as_os_str()))
+}
+
+/// We try to find the circom binary at `<CIRCOM_DIR>/circom`,
+/// where `<CIRCOM_DIR>` can be configured via the config file,
+/// a environment variable, or through a CLI argument, in that order.
+///
+/// We *do not* consider the case where the user already has some
+/// `circom` binary downloaded. The user will have two possibly
+/// conflicting circom binaries floating around. However, things
+/// should be kept separate as Lurk will never touch the user binary
+/// and the user should never manually call the Lurk Circom binary.
+///
+/// Whatever path is chosen, we then test if the `circom` binary
+/// exists. If it does, we return the path. Otherwise we download
+/// the binary to the location and return the path.
+fn get_circom_binary() -> Result<Command> {
+    let circom_path = circom_binary_path();
+
+    let output = Command::new(&circom_path).arg("--version").output();
+
+    let success = match output {
+        Ok(output) => {
+            // TODO: in future add back checksum check?
+            output.status.success()
+                && String::from_utf8_lossy(&output.stdout).contains(CIRCOM_VERSION)
+        }
+        Err(_) => false,
+    };
+
+    if success {
+        Ok(Command::new(circom_path))
```

Comment:
Nit: you’ve defined this object above, would it be possible to reuse this definition?

Reply:
the `.arg().output()` consumes the first